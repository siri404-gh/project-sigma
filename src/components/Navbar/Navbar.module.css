.toolbar {
  flex-direction: row;
  justify-content: space-between;
}
