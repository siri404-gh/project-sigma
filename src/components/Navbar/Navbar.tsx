import React, { useState, MouseEvent, FC } from 'react'

import {
  AccountCircle as AccountIcon,
  Menu as MenuIcon,
} from '@mui/icons-material'
import {
  AppBar,
  Avatar,
  Box,
  IconButton,
  Link as MuiLink,
  Menu,
  MenuItem,
  Toolbar,
  Typography,
} from '@mui/material'
import Link from 'next/link'

import styles from './Navbar.module.css'

export interface NavbarProps {
  avatarUrl?: string
  children?: React.ReactNode
  description?: string
  isUserLoggedIn?: boolean
  loginRedirectUrl?: string
  onMenuClick?: () => void
  onToolbarClick?: () => void
  title?: string
}

const Navbar: FC<NavbarProps> = props => {
  const {
    avatarUrl,
    children,
    isUserLoggedIn,
    loginRedirectUrl,
    onMenuClick,
    title,
  } = props

  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null)
  const onMenuButtonClick = (e: MouseEvent<HTMLButtonElement>) => {
    e.stopPropagation()
    onMenuClick?.()
  }
  const onAvatarClick = (e: MouseEvent<HTMLButtonElement>) => {
    e.stopPropagation()
    setAnchorEl(!anchorEl ? e.currentTarget : null)
  }

  const appBarColor = 'secondary'
  const otherColor = 'primary'

  return (
    <AppBar
      color={appBarColor}
      component='nav'
      elevation={0}
      position='static'
      sx={theme => ({
        // borderBottom: `solid 1px ${theme.palette.primary.main}`,
      })}>
      <Toolbar className={styles.toolbar}>
        <Link href='/' passHref>
          <MuiLink underline='none'>
            <Typography color={otherColor} component='h1' variant='h6'>
              {title}
            </Typography>
          </MuiLink>
        </Link>
        {children}
        <Box
          sx={{ display: 'flex', justifyContent: 'right', width: { sm: 200 } }}>
          <IconButton
            aria-controls='simple-menu'
            color={otherColor}
            onClick={onAvatarClick}
            aria-hidden>
            {isUserLoggedIn ? (
              <Avatar alt='avatar' src={avatarUrl} />
            ) : (
              <AccountIcon />
            )}
            <Menu
              anchorEl={anchorEl}
              onClose={onAvatarClick}
              open={Boolean(anchorEl)}
              keepMounted>
              {isUserLoggedIn ? (
                <Link href={'/api/auth/logout'} passHref>
                  <MenuItem>Logout</MenuItem>
                </Link>
              ) : (
                <Link
                  href={`/api/auth/login?redirectTo=${loginRedirectUrl}`}
                  passHref>
                  <MenuItem>Login</MenuItem>
                </Link>
              )}
            </Menu>
          </IconButton>
          <IconButton
            aria-label='menu'
            color={otherColor}
            onClick={onMenuButtonClick}
            sx={{ pr: 0 }}>
            <MenuIcon fontSize='large' />
          </IconButton>
        </Box>
      </Toolbar>
    </AppBar>
  )
}

export default Navbar
