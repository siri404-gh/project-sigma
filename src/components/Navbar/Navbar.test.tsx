import * as React from 'react'

import { render, screen } from '@testing-library/react'

import Navbar from '@/components/Navbar/Navbar'

describe('Navbar', () => {
  test('renders', () => {
    const { container } = render(<Navbar />)
    expect(container).toMatchSnapshot()
  })
  test('renders title', () => {
    render(<Navbar title='title' />)
    const titleDiv = screen.getByText(/title/i)
    expect(titleDiv).toHaveTextContent('title')
    expect(screen.getByRole('navigation')).toBeInTheDocument()
  })
})
