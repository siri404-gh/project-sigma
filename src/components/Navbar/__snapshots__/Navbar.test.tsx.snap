// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders 1`] = `
<div>
  <nav
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-v526sl-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-1p4tas-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular css-16ch7h7-MuiToolbar-root"
      >
        <div
          class="MuiBox-root css-k008qs"
        >
          <button
            aria-label="menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-cgbg31-MuiTypography-root-MuiLink-root"
            href="/"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1rhpbw5-MuiTypography-root"
            />
          </a>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
